internal/api/update_secrets_v1alpha1.go

```diff
@@ -0,0 +1,116 @@
+package api
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"connectrpc.com/connect"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+
+	kargoapi "github.com/akuity/kargo/api/v1alpha1"
+	svcv1alpha1 "github.com/akuity/kargo/pkg/api/service/v1alpha1"
+)
+
+func (s *server) UpdateSecrets(
+	ctx context.Context,
+	req *connect.Request[svcv1alpha1.UpdateSecretsRequest],
+) (*connect.Response[svcv1alpha1.UpdateSecretsResponse], error) {
+	// Check if secret management is enabled
+	if !s.cfg.SecretManagementEnabled {
+		return nil, connect.NewError(connect.CodeUnimplemented, errSecretManagementDisabled)
+	}
+
+	project := req.Msg.GetProject()
+
+	if err := validateFieldNotEmpty("project", project); err != nil {
+		return nil, err
+	}
+
+	name := req.Msg.GetName()
+
+	if err := validateFieldNotEmpty("name", name); err != nil {
+		return nil, err
+	}
+
+	secret := corev1.Secret{}
+	if err := s.client.Get(
+		ctx,
+		types.NamespacedName{
+			Namespace: project,
+			Name:      name,
+		},
+		&secret,
+	); err != nil {
+		return nil, connect.NewError(connect.CodeNotFound, fmt.Errorf("secret %s/%s not found", project, name))
```

Comment:
You don't need to be explicit about the not found error code here. The error handling for the entire API server knows how to translate Kubernetes client errors to an appropriate error code.